proximal/examples/test_deconv_sv_psf.py

```diff
@@ -0,0 +1,58 @@
+# Spatially Varying PSF Deconvolution Algorithm based on:
+# Flicker & Rigaut, 2005 https://doi.org/10.1364/JOSAA.22.000504
+
+import sys
+
+sys.path.append('../../')
+
+import proximal as px
+from proximal.lin_ops.black_box import LinOpFactory
+from scipy.signal import fftconvolve
+import numpy as np
+
+############################################################
+
+
+def my_factory(G, W, im_shape, n_psf: int):
+    assert np.all(G.shape == W.T.shape)
+    assert im_shape[0] * im_shape[1] == G.shape[0]
+    psf_mode = G[:, :n_psf].T.reshape((n_psf, *im_shape))
+    weight = W[:n_psf].reshape((n_psf, *im_shape))
+
+    def forward(im_src, im_sim):
+        assert np.all(im_src.shape == im_sim.shape)
+        im_sim[:] = 0
+        for i in range(n_psf):
+            im_sim[:] += fftconvolve(psf_mode[i],
+                                     im_src * weight[i],
+                                     mode='same')
+
+    def adjoint(im_src, im_sim):
+        assert np.all(im_src.shape == im_sim.shape)
+        im_sim[:] = 0
+        for i in range(n_psf):
+            im_sim += fftconvolve(im_src,
+                                  np.flipud(np.fliplr(psf_mode[i])),
+                                  mode='same') * weight[i]
+
+    return LinOpFactory(im_shape, im_shape, forward, adjoint)
+
+
+def cost_function(x, im, U, W, mu=0.01):
+    blur_op = my_factory(U, W, im.shape, 15)
+    im_blur = blur_op(x)
+    data_term = px.sum_squares(im_blur - im)
+    im_grad = px.grad(im_blur)
+    grad_term = mu * px.norm1(im_grad) + (1 - mu) * px.sum_squares(im_grad)
+    return data_term + grad_term + px.nonneg(im_blur)
+
+
+# Sample Data
+im = np.random.randn(128, 128)
+W = np.random.randn(20, np.prod(im.shape))
+U = np.random.randn(np.prod(im.shape), 20)
+
+# Solve
+x = px.Variable(im.shape)
+prob = px.Problem(cost_function(x, im, U, W))
+prob.solve(solver='pc', max_iters=2, verbose=True, x0=im.copy())
```

Comment:
I can actually merge the test script, so that it will show up in the examples folder. I just need a more realistic value of `W` and `U`. `width = 128` and `n_psf <= 20` can stay.